File: src/dotnet/Ansight/Pairing/Models/DeviceApplicationIconProfile.cs
namespace Ansight.Pairing.Models;

/// <summary>
/// Describes an application icon captured from the installed package or bundle.
/// </summary>
public sealed class DeviceApplicationIconProfile
{
    /// <summary>
    /// Image format code, such as png or jpeg.
    /// </summary>
    public string? Format { get; set; }

    /// <summary>
    /// MIME type for the encoded icon bytes.
    /// </summary>
    public string? MimeType { get; set; }

    /// <summary>
    /// Width of the encoded icon in pixels, when known.
    /// </summary>
    public int? Width { get; set; }

    /// <summary>
    /// Height of the encoded icon in pixels, when known.
    /// </summary>
    public int? Height { get; set; }

    /// <summary>
    /// Number of encoded bytes before base64 transport encoding.
    /// </summary>
    public long? ByteCount { get; set; }

    /// <summary>
    /// Base64-encoded image bytes.
    /// </summary>
    public string? DataBase64 { get; set; }
}

File: src/ios/Sources/AnsightCore/DeviceApplicationIconProfile.swift
import Foundation

#if canImport(Darwin)
import Darwin
#endif

#if canImport(UIKit)
import UIKit
#endif

public struct DeviceApplicationIconProfile: Sendable, Codable, Equatable {
    public var format: String?
    public var mimeType: String?
    public var width: Int?
    public var height: Int?
    public var byteCount: Int?
    public var dataBase64: String?

    public init(
        format: String? = nil,
        mimeType: String? = nil,
        width: Int? = nil,
        height: Int? = nil,
        byteCount: Int? = nil,
        dataBase64: String? = nil
    ) {
        self.format = format
        self.mimeType = mimeType
        self.width = width
        self.height = height
        self.byteCount = byteCount
        self.dataBase64 = dataBase64
    }
}
